lib/truffle/socket/basic_socket.rb

```diff
@@ -121,6 +121,7 @@ def getpeername
   end
 
   def send(message, flags, dest_sockaddr = nil)
+    message    = String(message)
```

Comment:
The `StringValue(message)` helper is typically used instead, and it seems more correct because that only tries `to_str`, and not `to_s`, which seems not accepted on CRuby:
```
> s1,s2 = UNIXSocket.pair
> obj = Object.new
> s1.send(obj, 0)
TypeError: no implicit conversion of Object into String
```